### ISSUE TITLE
run_every: time parameter has no effect

### ISSUE THREAD

**Issue by der-dag** [spacemanspiff2007/HABApp#210]
If I use example code from MQTT Rule example it is not working as expected:

`self.run_every(
            time=datetime.timedelta(seconds=60),
            interval=datetime.timedelta(seconds=30),
            callback=self.publish_rand_value
        )`

I would expect that self.publish_rand_value() is called first time in 90s and after that every 30s. But it is first time called after 30s. It seems the time parameter has no effect.
Same problem when using run_minutely, it is always started immediatly and not at a random seconds as documented.

**Comment by spacemanspiff2007** [comment]
I've made many changes and reworked the scheduler.
Would you be willing to try out the dev branch?
Be aware there are many changes so if possible try from another machine!

**Comment by TBail** [comment]
I tried it with the latest dev branch from yesterday and it seems to work now as you camm see from the log.

```
[2021-04-16 09:59:44,485] [                    Rules]     INFO | Energy management: New value = 1738.35m³.

[2021-04-16 10:00:00,208] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-16 10:02:04,377] [                    Rules]     INFO | Energy management: New value = 1738.360001m³.

[2021-04-16 10:04:10,113] [                    Rules]     INFO | Energy management: New value = 1738.379999m³.

[2021-04-16 10:07:04,185] [                    Rules]     INFO | Energy management: New value = 1738.4m³.

[2021-04-16 10:09:44,342] [                    Rules]     INFO | Energy management: New value = 1738.42m³.

[2021-04-16 10:12:10,338] [                    Rules]     INFO | Energy management: New value = 1738.43m³.

[2021-04-16 10:14:22,178] [                    Rules]     INFO | Energy management: New value = 1738.450001m³.

[2021-04-16 10:16:19,858] [                    Rules]     INFO | Energy management: New value = 1738.459999m³.

[2021-04-16 10:19:06,123] [                    Rules]     INFO | Energy management: New value = 1738.48m³.

[2021-04-16 10:21:38,223] [                    Rules]     INFO | Energy management: New value = 1738.5m³.

[2021-04-16 10:23:56,167] [                    Rules]     INFO | Energy management: New value = 1738.52m³.

[2021-04-16 10:26:00,197] [                    Rules]     INFO | Energy management: New value = 1738.530001m³.

[2021-04-16 10:28:52,557] [                    Rules]     INFO | Energy management: New value = 1738.549999m³.

[2021-04-16 11:13:46,684] [                    Rules]     INFO | Energy management: New value = 1738.82m³.

[2021-04-16 11:16:21,471] [                    Rules]     INFO | Energy management: New value = 1738.84m³.

[2021-04-16 11:18:42,100] [                    Rules]     INFO | Energy management: New value = 1738.85m³.

[2021-04-16 11:20:48,565] [                    Rules]     INFO | Energy management: New value = 1738.870001m³.

[2021-04-16 11:23:43,609] [                    Rules]     INFO | Energy management: New value = 1738.889999m³.

[2021-04-16 11:26:24,500] [                    Rules]     INFO | Energy management: New value = 1738.91m³.

[2021-04-16 11:28:51,223] [                    Rules]     INFO | Energy management: New value = 1738.93m³.

[2021-04-16 11:31:47,247] [                    Rules]     INFO | Energy management: New value = 1738.950001m³.

[2021-04-16 11:34:29,359] [                    Rules]     INFO | Energy management: New value = 1738.969999m³.

[2021-04-16 11:39:11,097] [                    Rules]     INFO | Energy management: New value = 1738.99m³.

[2021-04-16 11:48:52,888] [                    Rules]     INFO | Energy management: New value = 1739.0m³.

[2021-04-16 11:58:59,592] [                    Rules]     INFO | Energy management: New value = 1739.02m³.

[2021-04-16 12:00:00,212] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-16 12:01:11,670] [                    Rules]     INFO | Energy management: New value = 1739.030001m³.
```

Adding a second run.every brings another error

```
[2021-04-16 12:17:32,377] [             HABApp.Rules]    ERROR | Could not load /etc/openhab/habapp/rules/energymanagement.py!

[2021-04-16 12:17:32,380] [             HABApp.Rules]    ERROR | File "/etc/openhab/habapp/rules/energymanagement.py", line 44, in energymanagement.py

[2021-04-16 12:17:32,380] [             HABApp.Rules]    ERROR |     40           log.info(f'Energy management: Daily update.')

[2021-04-16 12:17:32,382] [             HABApp.Rules]    ERROR |      NaturalGasManager = <class '/etc/openhab/habapp/rules/energymanagement.py.NaturalGasManager'>

[2021-04-16 12:17:32,382] [             HABApp.Rules]    ERROR | File "/etc/openhab/habapp/rules/energymanagement.py", line 24, in __init__

[2021-04-16 12:17:32,384] [             HABApp.Rules]    ERROR |     23       self.run.every(datetime.now().replace(minute=0, second=0, microsecond=0), timedelta(hours=1), self.update_hourly_energy_consumption)

[2021-04-16 12:17:32,384] [             HABApp.Rules]    ERROR | --> 24       self.run.every(datetime.now().replace(hour=0, minute=0, second=0, microsecond=0), timedelta(days=1), self.update_daily_energy_consumption)

[2021-04-16 12:17:32,385] [             HABApp.Rules]    ERROR |      self = </etc/openhab/habapp/rules/energymanagement.py.NaturalGasManager object at 0x7fbb642fe4f0>

[2021-04-16 12:17:32,385] [             HABApp.Rules]    ERROR |      self.update_hourly_energy_consumption = <method 'NaturalGasManager.update_hourly_energy_consumption' of </etc/openhab/habapp/rules/energymanagement.py.NaturalGasManager object at 0x7fbb642fe4f0> energymanagement.py:35>

[2021-04-16 12:17:32,385] [             HABApp.Rules]    ERROR |      self.update_daily_energy_consumption = <method 'NaturalGasManager.update_daily_energy_consumption' of </etc/openhab/habapp/rules/energymanagement.py.NaturalGasManager object at 0x7fbb642fe4f0> energymanagement.py:39>

[2021-04-16 12:17:32,389] [             HABApp.Rules]    ERROR |      self._rule._get_cb_name = <method 'Rule._get_cb_name' of </etc/openhab/habapp/rules/energymanagement.py.NaturalGasManager object at 0x7fbb642fe4f0> rule.py:249>

[2021-04-16 12:17:32,402] [             HABApp.Rules]    ERROR | File "/etc/openhab/habapp/rules/energymanagement.py", line 44, in energymanagement.py

[2021-04-16 12:17:32,402] [             HABApp.Rules]    ERROR | File "/etc/openhab/habapp/rules/energymanagement.py", line 24, in __init__

[2021-04-16 12:17:32,402] [             HABApp.Rules]    ERROR |     self.run.every(datetime.now().replace(hour=0, minute=0, second=0, microsecond=0), timedelta(days=1), self.update_daily_energy_consumption)

```

Code ist like that
```
class NaturalGasManager(HABApp.Rule):
    """Rule class to manage (calculate) natural gas consumption"""

    def __init__(self):
        """
        Description
        -----------
        Initialize the class with alle required stuff like eventlisteners etc.
        """
        super().__init__()

        self.run.every(datetime.now().replace(minute=0, second=0, microsecond=0), timedelta(hours=1), self.update_hourly_energy_consumption)
        self.run.every(datetime.now().replace(hour=0, minute=0, second=0, microsecond=0), timedelta(days=1), self.update_daily_energy_consumption)

        self.energy_item = NumberItem.get_item(f'RH_BM_LD_GasMeter_GasEnergyCounter')
        self.energy_item.listen_event(self.update_energy_counter, ValueChangeEvent)

        log.info(f'Energy management: Initialized.')

    def update_energy_counter(self, event: ValueChangeEvent):
        log.info(f'Energy management: New value = {event.value}m³.')
        pass

    def update_hourly_energy_consumption(self):
        log.info(f'Energy management: Hourly update')
        pass

    def update_daily_energy_consumption(self):
        log.info(f'Energy management: Daily update.')
        pass```

**Comment by TBail** [comment]
Interessting effect. As you see i added the microseconds option to the replace function. Now the function is calles many times.
```
[2021-04-16 16:53:56,550] [                    Rules]     INFO | Energy management: New value = 1740.200001m³.

[2021-04-16 16:56:19,865] [                    Rules]     INFO | Energy management: New value = 1740.219999m³.

[2021-04-16 16:58:29,260] [                    Rules]     INFO | Energy management: New value = 1740.23m³.

[2021-04-16 17:00:00,006] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-16 17:00:00,010] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-16 17:00:00,013] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-16 17:00:00,016] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-16 17:00:00,019] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-16 17:00:00,022] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-16 17:00:00,210] [                    Rules]     INFO | Energy management: Hourly update```

**Comment by spacemanspiff2007** [comment]
There was still a bug in the scheduler which should be fixed now.
Could you please try again but before do
``pip install --upgrade eascheduler``
which should install eascheduler 0.0.8 and restart habapp

**Comment by TBail** [comment]
Updated the eascheduler, but that didn't fix the problem

```
[2021-04-17 13:53:30,090] [                    Rules]     INFO | Energy management: New value = 1748.799999m³.

[2021-04-17 13:55:47,298] [                    Rules]     INFO | Energy management: New value = 1748.82m³.

[2021-04-17 14:00:00,009] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,012] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,017] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,023] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,026] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,029] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,009] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,012] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,017] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,023] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,026] [                    Rules]     INFO | Energy management: Hourly update

[2021-04-17 14:00:00,029] [                    Rules]     INFO | Energy management: Hourly update```

**Comment by spacemanspiff2007** [comment]
If I try with the latest scheduler and latest HABApp I get a ``FirstRunNotInTheFutureError`` error (which is correct).
Maybe something while updating went wrong? Have you restarted HABApp?`

If I modify the rule I get the correct output:
```python
        print(eascheduler.__version__)

        a = self.run.every(time((datetime.now() + timedelta(hours=1)).hour), timedelta(hours=1), self.update_hourly_energy_consumption)
        print(a.get_next_run())

        b = self.run.every(time(0), timedelta(days=1), self.update_daily_energy_consumption)
        print(b.get_next_run())
```

```
0.0.8
2021-04-17T16:00:00
2021-04-18T00:00:00
```

**Comment by TBail** [comment]
After updating to the latest dev branch and fixing some erreor in my script with help from @spacemanspiff2007 it is now working

**Comment by spacemanspiff2007** [comment]
Fixed with 0.30.0
